web-ts/src/routes/LocalNewsPage.tsx

```diff
@@ -12,24 +30,120 @@ type PropsType = {
   languageModel: LanguageModel
 } & RouteComponentProps<{ cityCode: string; languageCode: string; newsId?: string }>
 
-const LocalNewsPage = ({ match, cityModel, location }: PropsType) => {
-  const { languageCode } = match.params
-  const toolbar = (openFeedback: (rating: FeedbackRatingType) => void) => (
-    <LocationToolbar openFeedbackModal={openFeedback} viewportSmall={false} />
-  )
+const LocalNewsPage = ({ match, cityModel, languages, location }: PropsType): ReactElement => {
+  const { cityCode, languageCode, newsId } = match.params
+  const pathname = normalizePath(location.pathname)
+  const history = useHistory()
+  const formatter = useContext(DateFormatterContext)
+  const { t } = useTranslation('news')
+  const viewportSmall = false
+
+  const requestLocalNews = useCallback(async () => {
+    return createLocalNewsEndpoint(cmsApiBaseUrl).request({ city: cityCode, language: languageCode })
+  }, [cityCode, languageCode])
+  const { data: localNews, loading, error: newsError } = useLoadFromEndpoint(requestLocalNews)
+
+  const newsModel = newsId && localNews?.find(it => it.id.toString() === newsId)
+
+  const renderLocalNewsListItem = (localNewsItem: LocalNewsModel) => {
+    const { id, title, message, timestamp } = localNewsItem
+    return (
+      <NewsListItem
+        title={title}
+        content={message}
+        timestamp={timestamp}
+        key={id}
+        link={createPath(LOCAL_NEWS_ROUTE, { cityCode, languageCode, newsId: id })}
+        t={t}
+        formatter={formatter}
+        type={LOCAL_NEWS_TYPE}
+      />
+    )
+  }
+
+  // Language change is not possible between local news detail views because we don't know the id of other languages
+  const languageChangePaths = languages.map(({ code, name }) => ({
+    path: newsId ? null : createPath(LOCAL_NEWS_ROUTE, { cityCode, languageCode: code }),
```

Comment:
Still wouldn't it be more intuitive to return to `createPath(LOCAL_NEWS_ROUTE, { cityCode, languageCode: code })`?

Reply:
That would differ from the behavior we currently have in both web and native, so I think this should happen in a separate issue if we want it. Maybe you are right though, but not really sure about that...